Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -678,28 +764,50 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
     }
   }
 
-  /// Enum to enable using
-  /// [Automatic Persisted Queries (APQs)](https://www.apollographql.com/docs/apollo-server/performance/apq)
-  /// with your generated operations.
-  ///
-  /// APQs are an Apollo Server feature. When using Apollo iOS to connect to any other GraphQL server,
-  /// `APQConfig` should be set to `.disabled`
-  public enum APQConfig: String, Codable, Equatable {
-    /// The default value. Disables APQs.
-    /// The operation document is sent to the server with each operation request.
-    case disabled
+  public struct OperationDocumentFormat: OptionSet, Codable, Equatable {
+    /// Include the GraphQL source document for the operation in the generated operation models.
+    public static let definition = Self(rawValue: 1)
+    /// Include the computed operation identifier hash for use with persisted queries
+    /// or [Automatic Persisted Queries (APQs)](https://www.apollographql.com/docs/apollo-server/performance/apq).
+    public static let operationId = Self(rawValue: 1 << 1)
 
-    /// Automatically persists your operations using Apollo Server's
-    /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
-    case automaticallyPersist
+    public var rawValue: UInt8
+    public init(rawValue: UInt8) {
+      self.rawValue = rawValue
+    }
 
-    /// Provides only the `operationIdentifier` for operations that have been previously persisted
-    /// to an Apollo Server using
-    /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
-    ///
-    /// If the server does not recognize the `operationIdentifier`, the operation will fail. This
-    /// method should only be used if you are manually persisting your queries to an Apollo Server.
-    case persistedOperationsOnly
+    // MARK: Codable
+
+    public enum CodingKeys: String, CodingKey {
+      case definition
+      case operationId
+    }
+
+    public init(from decoder: Decoder) throws {
+      self = OperationDocumentFormat(rawValue: 0)
+
+      var container = try decoder.unkeyedContainer()
+      while !container.isAtEnd {
+        let value = try container.decode(String.self)
+        switch CodingKeys(rawValue: value) {
+        case .definition:
+          self.insert(.definition)
+        case .operationId:
+          self.insert(.operationId)
+        default: continue
+        }
+      }
+    }
+
+    public func encode(to encoder: Encoder) throws {
```

Comment:
Do we need the `encode` function or does the default implementation result in some obscure JSON formatting for the value?

Reply:
The default encoded value for an OptionSet is just the raw value (ie. an `Int`). I'm doing manual Codable here so that the JSON config file can have these look like an array of values.

Reply:
Yes, that'll be much nicer.